llama_index/output_parsers/selection.py

```diff
@@ -48,27 +48,35 @@ class Answer(DataClassJsonMixin):
 
 class SelectionOutputParser(BaseOutputParser):
     def _marshal_llm_to_json(self, output: str) -> str:
-        """Extract a valid JSON array from a string.
-        Extracts a substring that represents a valid JSON array.
+        """Extract a valid JSON object or array from a string.
+        Extracts a substring that represents a valid JSON object or array.
 
         Args:
-            output: A string that may contain a valid JSON array surrounded by
+            output: A string that may contain a valid JSON object or array surrounded by
             extraneous characters or information.
 
         Returns:
-            A string representing a valid JSON array.
+            A string representing a valid JSON object or array.
 
         """
         output = output.strip()
-        left = output.find("[")
-        right = output.rfind("]")
+        if output[0] == '{':
```

Comment:
This feels a bit brittle personally, now that I look at this again, as sometimes the LLM may have extra output before the JSON starts

Maybe a better option is to find the `left` using both `[` and `{`, and whichever one is further left changes the symbol you use to find the `right`? Let me know if this makes sense

```python
left_sqaure = output.find("[")
left_brace = output.find("{")

if left_square < left_brace:
  <find right square>
else:
  <find right brace>
```